### DISCUSSION TITLE
installing Storybook 7.5.3 with vite (TypeError: createFilter is not a function)

### DISCUSSION THREAD

**Discussion by quintonjs** [storybookjs/storybook#24791] [Help] [answered]
Hi there,

I have a Vite solution up and running and then tried to install Storybook using the following:

`npx storybookj init --builder=vite`

it got so far and then exited out of the process with the return error message:

Running Storybook
yarn run v1.19.1
warning package.json: No license field
$ storybook dev -p 6006 --initial-path=/onboarding --quiet
@storybook/cli v7.5.3

info => Starting manager..
info => Starting preview..
=> Failed to build the preview
TypeError: createFilter is not a function
    at injectExportOrderPlugin (./node_modules/@storybook/builder-vite/dist/index.js:1:6848)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async pluginConfig (./node_modules/@storybook/builder-vite/dist/index.js:159:5991)
    at async commonConfig (./node_modules/@storybook/builder-vite/dist/index.js:159:5557)
    at async createViteServer (./node_modules/@storybook/builder-vite/dist/index.js:159:10288)
    at async Module.start (./node_modules/@storybook/builder-vite/dist/index.js:159:12520)
    at async storybookDevServer (./node_modules/@storybook/core-server/dist/index.js:104:2460)
    at async buildDevStandalone (./node_modules/@storybook/core-server/dist/index.js:119:3007)
    at async withTelemetry (./node_modules/@storybook/core-server/dist/index.js:103:3903)
    at async dev (./node_modules/@storybook/cli/dist/generate.js:473:401)

WARN Broken build, fix the error above.
WARN You may need to refresh the browser.

✔ Would you like to help improve Storybook by sending anonymous crash reports? … no
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

My package.json is as follows:

```
{
  "name": "vite-react-typescript-starter",
  "version": "0.0.0",
  "scripts": {
    "dev": "vite",
    "build": "tsc && vite build",
    "preview": "vite preview",
    "test": "jest",
    "storybook": "storybook dev -p 6006",
    "build-storybook": "storybook build"
  },
  "dependencies": {
    "@types/jest": "29.2.4",
    "class-variance-authority": "^0.7.0",
    "jest": "29.3.1",
    "jest-environment-jsdom": "^29.3.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-router-dom": "^6.18.0",
    "swr": "2.0.0",
    "tailwind-merge": "^2.0.0",
    "zustand": "^4.4.6"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.20.2",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@storybook/addon-essentials": "^7.5.3",
    "@storybook/addon-interactions": "^7.5.3",
    "@storybook/addon-links": "^7.5.3",
    "@storybook/addon-onboarding": "^1.0.8",
    "@storybook/blocks": "^7.5.3",
    "@storybook/react": "^7.5.3",
    "@storybook/react-vite": "^7.5.3",
    "@storybook/testing-library": "^0.2.2",
    "@testing-library/dom": "8.0.0",
    "@testing-library/jest-dom": "5.14.1",
    "@testing-library/react": "11.0.4",
    "@testing-library/react-hooks": "5.0.3",
    "@testing-library/user-event": "12.3.0",
    "@types/react": "^17.0.33",
    "@types/react-dom": "^17.0.10",
    "@vitejs/plugin-react": "^1.0.7",
    "autoprefixer": "^10.4.13",
    "identity-obj-proxy": "^3.0.0",
    "postcss": "^8.4.20",
    "prettier": "^3.0.3",
    "prettier-plugin-tailwindcss": "^0.5.6",
    "storybook": "^7.5.3",
    "tailwindcss": "^3.2.4",
    "typescript": "^4.4.4",
    "vite": "^2.6.13"
  },
  "jest": {
    "testEnvironment": "jest-environment-jsdom",
    "moduleNameMapper": {
      "\\.(css|less|scss|sass)$": "identity-obj-proxy"
    }
  }
}
```

**Comment by valentinpalkovic** [comment, marked as answer]
The minimal supported version of Vite is 3. Please update your project to use at least Vite v3. Otherwise, Storybook does not support your setup.
